packages/protobuf/src/private/util-common.ts

```diff
@@ -164,7 +164,10 @@ export function makeUtilCommon(): Omit<Util, "newFieldList" | "initFields"> {
             throw new Error(`oneof cannot contain ${s.kind}`);
           case "map":
             const keys = Object.keys(va);
-            if (keys.some((k) => vb[k] === undefined)) {
+            if (
+              keys.some((k) => vb[k] === undefined) ||
+              Object.keys(vb).some((k) => va[k] === undefined)
+            ) {
               return false;
             }
```

Comment:
Thinking about this a bit, I'm not sure we gain much from checking for `undefined` in the first place. We can just as well rely on `scalarEquals()` and `messageType.equals()`, which already handle `undefined`.

```diff
-           const keys = Object.keys(va);
-           if (
-            keys.some((k) => vb[k] === undefined) ||
-             Object.keys(vb).some((k) => va[k] === undefined)
-           ) {
-             return false;
-           }
+           const keys = Object.keys(va).concat(Object.keys(vb));
```

Do you mind updating this?